doc/docusaurus/docs/glossary.md

```diff
@@ -0,0 +1,166 @@
+The Valentine hard fork is an example of an intra-era hard fork.
+
+### Ledger Era
+
+A ledger era marks a specific period where new features are added to the Cardano ledger, for instance
+
+- The Alonzo era, which followed the Alonzo hard fork, introduced smart contracts and Plutus V1.
+- The Babbage era, which followed the Vasil hard fork, introduced Plutus V2 along with features such as reference scripts and inline datum.
+- The Conway era, which followed the Chang hard fork, introduced Plutus V3 and features such as governance actions and voting.
+
+A hard fork is required to introduce a new ledger era, but a hard fork does not necessarily introduce a new ledger era.
+
+### Ledger Language Version
+
+This is what "Plutus V1", "Plutus V2", "Plutus V3" refer to.
+See [Different Notions of Version](./essential-concepts/versions) and [Plutus Ledger Language Version](./working-with-scripts/ledger-language-version).
+
+### Minting Policy Script
+
+A Plutus script which must be satisfied in order for a transaction to mint tokens of the corresponding currency.
+
+### Off-chain Code
+
+Code executed by individual applications rather than Cardano nodes.
+This includes functions like building, signing, and submitting transactions.
+Off-chain code can be developed using libraries like [Mesh](https://meshjs.dev/), [PyCardano](https://pycardano.readthedocs.io/en/latest/) and [Cardano API](https://github.com/IntersectMBO/cardano-api).
+
+### On-chain Code
+
+Code executed directly on the Cardano blockchain by each participating node.
+The main function of on-chain code is for nodes to validate transactions, such as checking if a transaction is permitted to spend a UTXO or mint a token.
+On-chain code is usually written in a high level language like Plutus Tx, and compiled into Untyped Plutus Core (UPLC).
+The Cardano node includes an evaluator that runs UPLC programs.
+
+### The Plugin
+
+The compiler from Plutus Tx to Untyped Plutus Core, which is implemented as a GHC plugin.
+
+### Plutus
+
+The term "Plutus" can refer to Untyped Plutus Core, Typed Plutus Core, or, prior to its renaming to Plinth, Plutus Tx.
+To avoid ambiguity, it is advisable not to use "Plutus" on its own.
+
+### Plutus Core
+
+The term "Plutus Core" can refer either to Untyped Plutus Core or Typed Plutus Core, depending on the context.
+To avoid confusion, it is recommended to use UPLC for Untyped Plutus Core and TPLC for Typed Plutus Core.
+
+### Plutus IR
+
+An intermediate language that compiles to Plutus Core.
+See [Plutus Core and Plutus Tx](./essential-concepts/plutus-core-and-plutus-tx.md).
+
+### Plutus Metatheory
+
+The formalization of typed and untyped Plutus Core.
+In the future we may add Plutus IR to the formalization.
+It is "meta" in the sense that it seeks to answer the questions of whether the language and its semantics are correct, rather than whether a particular program is correct.
```

Comment:
What does it mean for the "language and its semantics" to be correct? I think it would be better to say something more in line with:
```suggestion
It is "meta" in the sense that it is a framework for reasoning about the Plutus Core languages themselves.
```

Reply:
This was directly copied from plutus-metatheory/README.md, but your suggestion sounds fine.